src/nni_manager/training_service/reusable/remote/machineScheduler.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as assert from 'assert';
+import { RemoteMachineMeta } from '../../remote_machine/remoteMachineData';
+import { getLogger, Logger } from '../../../common/log';
+
+/**
+ * A simple remote machine scheduler implementation
+ */
+export class MachineScheduler {
+
+    private readonly remoteMachineMetaOccupiedMap: Map<RemoteMachineMeta, boolean>;
+    private readonly log: Logger;
+
+    constructor(remoteMachineMetaList: RemoteMachineMeta[]) {
+        assert(remoteMachineMetaList.length > 0);
+        this.remoteMachineMetaOccupiedMap = new Map<RemoteMachineMeta, boolean>();
```

Comment:
It looks only maintain a map. so it doesn't need to be a separated class. Should be in enviornment service.

Reply:
keep a extra class is more clearly, while it might cause a little complex in code. Removed in current version, could add them back if have more requirement in the future.